### PR TITLE
Added data source DB model

### PR DESCRIPTION
**Pull request by berggren**
With the new era of composite indices, and append-able timelines we need a way to track the origin of the data uploaded, and how it was collected. There can be multiple imports from various CSV files to a single timeline.

This PR adds the DB model, and functionality to add this metadata to every new timeline created.

### CODE REVIEW

File: timesketch/api/v1/resources/upload.py [hunk 1]
```diff
@@ -385,8 +408,6 @@ def post(self):
         utils.update_sketch_last_activity(sketch)
 
         # TODO: Read the data provider and save the information to the DB.
```
Comment (kiddinn) [inline timesketch/api/v1/resources/upload.py:410]:
```suggestion
     
```

**Review by kiddinn — COMMENTED** [review]
only minor suggestions, otherwise LGTM

File: timesketch/api/v1/resources/upload.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 from timesketch.models.sketch import SearchIndex
 from timesketch.models.sketch import Sketch
 from timesketch.models.sketch import Timeline
-
+from timesketch.models.sketch import DataSource
```
[the remark this answers is not in the document]
Comment (berggren) [reply timesketch/api/v1/resources/upload.py:39]: done

**Review by kiddinn — APPROVED** [review]
LGTM

**Pull request merged**